syncer/syncer.go

```diff
@@ -1404,16 +1436,37 @@ func (s *Syncer) syncDML(
 
 // Run starts running for sync, we should guarantee it can rerun when paused.
 func (s *Syncer) Run(ctx context.Context) (err error) {
-	tctx := s.tctx.WithContext(ctx)
+	runCtx, runCancel := context.WithCancel(context.Background())
```

Comment:
```suggestion
	runCtx, runCancel := context.WithCancel(context.Background())
	defer runCancel()
```

Reply:
That's the reason the CI `handle_error_3` failed.